src/deepsparse/server/system_logging.py

```diff
@@ -0,0 +1,121 @@
+
+        - for other system data groups, the SystemLoggingManager will call an
+            internal method of the `Pipeline` to log the data generated by the
+            pipeline (e.g. to obtain the inference latency of the pipeline). This
+            will bypass the SystemLoggingManager object - the data will flow from
+            the `Pipeline` directly to the `server_logger` through the `Pipeline`'s
+            `log` method.
+
+    :param system_group_names: a list of system group names that define the scope
+        of system metrics that are logged.
+        E.g. ["resource_utilization", "request_details""]
+
+
+    """
+
+    def __init__(self, system_group_names: List[str]):
+        self.system_group_names = system_group_names
+
+    @classmethod
+    def from_server_config(cls, server_config: ServerConfig) -> "SystemLoggingManager":
+        """
+        Factory method that creates a SystemLoggingManager
+        object from a ServerConfig object.
+        :param server_config: a ServerConfig object
+        :return: a SystemLoggingManager object
+        """
+        system_logging_config = server_config.system_logging
+        system_group_names = extract_system_group_data(system_logging_config).keys()
+        return cls(system_group_names)
+
+    def log(self, pipeline: Pipeline, **kwargs: Dict[str, Any]):
+        """
+        Logs system metrics for the pipeline.
+
+        :param pipeline: a Pipeline object
+        :param kwargs: a dictionary of additional data to be logged
+        """
+        if not pipeline.logger:
+            # no server_logger is set for the pipeline
+            # no system metrics will be logged
+            return
+        for group_name in self.system_group_names:
+            # multiple switch statements that are called
+            # based on the available system group name
+            if group_name == "resource_utilization":
+                self.log_resource_utilization(pipeline)
+
+            elif group_name == "request_details":
+                self.log_request_details(pipeline, **kwargs)
+
+            else:
+                raise ValueError(f"Unknown system group name: {group_name}")
+
+    def log_resource_utilization(self, pipeline: Pipeline):
+        """
+        Scope for 1.4:
+        - CPU utilization overall
+        - Memory available overall
+        - Memory used overall (shall we continuously log this?
+          this will be a constant value in time)
+        - Number of core used by the pipeline
+        """
+        raise NotImplementedError
+
+    def log_request_details(self, pipeline: Pipeline, **kwargs: Dict[str, Any]):
+        """
+        Scope for 1.4:
+        - Number of Successful Requests
+        (binary events, 1 or 0 per invocation of an endpoint)
+        - Batch size
+        - Number of Inferences (
+        number of successful inferences times the respective batch size)
+        """
+        raise NotImplementedError
```

Comment:
i think this set of changes would be a lot simpler if we just explicitly put these functions in the server.py file as standalone functions/methods and don't have this additional object.

e.g.
```python
def _predict(...):
    log_request_details(...)
    log_resource_utilization(...)
   ...
```

Reply:
agreed overall. I like the structure here @dbogunowicz, but looking through it looks like the class isn't keeping much state so could be restructured to a helper function

Reply:
Yes, this is true for now (1.4 release). To obey the KISS principle for this release, I would employ helper function. If the scope of system logging potentially increases in the future and we need some more complexity, we can always default to a class object.